go/compress.go

```diff
@@ -106,45 +112,62 @@ func compressStep(step *InnerOp, lookup *[]*InnerOp, registry map[string]int32)
 	return num
 }
 
-func decompress(comp *CompressedBatchProof) *BatchProof {
+func decompress(comp *CompressedBatchProof) (*BatchProof, error) {
 	lookup := comp.LookupInners
 
 	var entries []*BatchEntry
 
 	for _, centry := range comp.Entries {
-		entry := decompressEntry(centry, lookup)
+		entry, err := decompressEntry(centry, lookup)
+		if err != nil {
+			return nil, err
+		}
 		entries = append(entries, entry)
 	}
 
 	return &BatchProof{
 		Entries: entries,
-	}
+	}, nil
 }
 
-func decompressEntry(entry *CompressedBatchEntry, lookup []*InnerOp) *BatchEntry {
+func decompressEntry(entry *CompressedBatchEntry, lookup []*InnerOp) (*BatchEntry, error) {
 	if exist := entry.GetExist(); exist != nil {
+		decompressedExist, err := decompressExist(exist, lookup)
+		if err != nil {
+			return nil, err
+		}
 		return &BatchEntry{
 			Proof: &BatchEntry_Exist{
-				Exist: decompressExist(exist, lookup),
+				Exist: decompressedExist,
 			},
-		}
+		}, nil
 	}
 
 	non := entry.GetNonexist()
+	decompressedLeft, err := decompressExist(non.Left, lookup)
+	if err != nil {
+		return nil, err
+	}
+
+	decompressedRight, err := decompressExist(non.Right, lookup)
+	if err != nil {
+		return nil, err
+	}
+
 	return &BatchEntry{
 		Proof: &BatchEntry_Nonexist{
 			Nonexist: &NonExistenceProof{
 				Key:   non.Key,
-				Left:  decompressExist(non.Left, lookup),
-				Right: decompressExist(non.Right, lookup),
+				Left:  decompressedLeft,
+				Right: decompressedRight,
 			},
 		},
-	}
+	}, nil
 }
 
-func decompressExist(exist *CompressedExistenceProof, lookup []*InnerOp) *ExistenceProof {
+func decompressExist(exist *CompressedExistenceProof, lookup []*InnerOp) (*ExistenceProof, error) {
 	if exist == nil {
-		return nil
+		return nil, nil
```

Comment:
tests broke with an error return here. I don't have a great understanding of when this case arises? Is there a comment we can add to elaborate on it? @AdityaSripal

Reply:
If a nonexistence proof is proving the leftmost key wouldn't the leftExist be nil? So decompressExist returns nil and we don't error out on line 149

Reply:
ah yes, thank you! That makes sense to me, added a comment